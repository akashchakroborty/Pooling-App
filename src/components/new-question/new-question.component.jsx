import React, { useState, useEffect } from "react";
import {
  ModelContainer,
  AddNewButton,
  SuccessMessage,
  ErrorMessage,
  CloseButton
} from "./new-question.styles";
import { connect } from "react-redux";

import ModelForm from "../model-form/model-form.component";

import { createStructuredSelector } from "reselect";
import { selectisNewQuestionAdded } from "../../redux/questions-list/questions-list.selector";

ModelContainer.setAppElement("#root");

const NewQuestion = ({ newQuestionAddedSuccess }) => {
  const [modalIsOpen, setModalIsOpen] = useState(false);

  const [visible, setVisible] = useState(true);

  useEffect(() => {
    setTimeout(() => setVisible(false), 5000);
  }, []);

  const openModal = () => {
    setModalIsOpen(true);
  };
  const closeModal = () => {
    setModalIsOpen(false);
  };

  return (
    <div>
      {visible &&
      newQuestionAddedSuccess &&
      newQuestionAddedSuccess === "success" ? (
        <SuccessMessage>New question created successfully!</SuccessMessage>
      ) : null}
      {visible &&
      newQuestionAddedSuccess &&
      newQuestionAddedSuccess === "failure" ? (
        <ErrorMessage>
          Error occurred while creating new question !
        </ErrorMessage>
      ) : null}
      <AddNewButton onClick={openModal}>Add New Question</AddNewButton>

      <ModelContainer isOpen={modalIsOpen} onRequestClose={closeModal}>
        <ModelForm />
        <CloseButton onClick={closeModal}>Close</CloseButton>
      </ModelContainer>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  newQuestionAddedSuccess: selectisNewQuestionAdded
});

export default connect(mapStateToProps)(NewQuestion);
