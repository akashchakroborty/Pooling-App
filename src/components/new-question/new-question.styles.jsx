import styled from "styled-components";
import Modal from "react-modal";

export const ModelContainer = styled(Modal)`
  max-width: 80vw;
  width: 80%;
  text-align: center;
  position: absolute;
  top: 43%;
  left: 10%;
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background: #0d223f;
`;

export const AddNewButton = styled.button`
  margin: 0 auto;
  margin: 1rem auto;
  display: flex;
`;

export const SuccessMessage = styled.div`
  margin: 1rem 0;
  color: #0d223f;
  text-align: center;
  font-size: 20px;
`;

export const ErrorMessage = styled.div`
  margin: 1rem 0;
  color: red;
  text-align: center;
  font-size: 20px;
`;

export const CloseButton = styled.button`
  margin: 1rem 0;
`;
