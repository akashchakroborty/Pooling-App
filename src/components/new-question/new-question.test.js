import React from "react";
import { shallow } from "enzyme";
import { NewQuestion } from "./new-question.component";
import {
  AddNewButton,
  CloseButton,
  SuccessMessage,
  ErrorMessage
} from "./new-question.styles";

describe("NewQuestion component", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      newQuestionAddedSuccess: false
    };

    wrapper = shallow(<NewQuestion {...mockProps} />);
  });

  it("should render NewQuestion component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call openModal when AddNewButton is clicked", () => {
    wrapper.find(AddNewButton).simulate("click");
    expect.any(Function);
  });

  it("should call closeModal when CloseButton is clicked", () => {
    wrapper.find(CloseButton).simulate("click");
    expect.any(Function);
  });

  it("should render SuccessMessage if newQuestionAddedSuccess is success", () => {
    const mockProps = {
      newQuestionAddedSuccess: "success"
    };

    const newWrapper = shallow(<NewQuestion {...mockProps} />);
    expect(newWrapper.exists(SuccessMessage)).toBe(true);
  });

  it("should render ErrorMessage if newQuestionAddedSuccess is failure", () => {
    const mockProps = {
      newQuestionAddedSuccess: "failure"
    };

    const newWrapper = shallow(<NewQuestion {...mockProps} />);
    expect(newWrapper.exists(ErrorMessage)).toBe(true);
  });
});
