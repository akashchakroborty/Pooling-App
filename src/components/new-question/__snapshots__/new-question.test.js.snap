// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion component should render NewQuestion component 1`] = `ShallowWrapper {}`;
